agent/src/main/java/com/newrelic/agent/android/stores/SharedPrefsEventStore.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2022-present New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.newrelic.agent.android.stores;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.newrelic.agent.android.analytics.AnalyticsEvent;
+import com.newrelic.agent.android.analytics.AnalyticsEventStore;
+import com.newrelic.agent.android.crash.Crash;
+import com.newrelic.agent.android.logging.AgentLog;
+import com.newrelic.agent.android.logging.AgentLogManager;
+import com.newrelic.agent.android.metric.MetricNames;
+import com.newrelic.agent.android.stats.StatsEngine;
+import com.newrelic.agent.android.util.SafeJsonPrimitive;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class SharedPrefsEventStore extends SharedPrefsStore implements AnalyticsEventStore {
+    private static final AgentLog log = AgentLogManager.getAgentLog();
+    private static final String STORE_FILE = "NREventStore";
+
+    public SharedPrefsEventStore(Context context) {
+        this(context, STORE_FILE);
+    }
+
+    public SharedPrefsEventStore(Context context, String storeFilename) {
+        super(context, storeFilename);
+    }
+
+    @Override
+    public boolean store(AnalyticsEvent event) {
+        synchronized (this) {
+            try {
+                final JsonObject jsonObj = event.asJsonObject();
+                String eventJson = jsonObj.toString();
+
+                // events should be stored synchronously, since the app is terminating
+                SharedPreferences.Editor editor = this.sharedPrefs.edit();
+                editor.putString(event.getEventUUID(), eventJson);
+
+                StatsEngine.get().inc(MetricNames.SUPPORTABILITY_EVENT_SIZE_UNCOMPRESSED, eventJson.length());
```

Comment:
For supportability metrics, use `StatsEngine.SUPPORTABILITY.inc(...)`as its decorator will also create metrics based on framework.